### PR TITLE
Solve issue 43, Fix default dataset

### PR DESCRIPTION
**Pull request by alonme**
Author:    alonme <alon.menczer@gmail.com>

### CODE REVIEW

File: pybigquery/sqlalchemy_bigquery.py [hunk 1]
```diff
@@ -267,47 +268,61 @@ def __init__(
...
+        if dataset_id:
+            # If project_id is missing, use defualt project_id
+            if not project_id:
+                _, project_id = auth.default()
 
-        self.arraysize = self.arraysize or arraysize
-        self.location = location or self.location
-        self.credentials_path = credentials_path or self.credentials_path
-        self.dataset_id = dataset_id
+            job_config.default_dataset = f'{project_id}.{dataset_id}'
```
Comment (tswast) [inline pybigquery/sqlalchemy_bigquery.py]: What versions of Python does `pybigquery` want to support? If we need to support Python 3.5, I don't think we can use fstrings.

File: pybigquery/sqlalchemy_bigquery.py [hunk 2]
```diff
@@ -267,47 +268,61 @@ def __init__(
...
-            credentials = credentials.with_scopes(scopes)
-            client = bigquery.Client(
-                project=self.credentials_info.get('project_id'),
+        credentials = credentials.with_scopes(scopes)
+
+        self._add_default_dataset_to_job_config(default_query_job_config,
+                                        credentials.project_id, self.dataset_id)
+
+        return bigquery.Client(
+                project=credentials.project_id,
```
Comment (tswast) [inline pybigquery/sqlalchemy_bigquery.py]: Note: there are some kinds of credentials that don't have a `project_id` property. e.g. if someone uses `gcloud application-default login`.

Edit: Looks like it's not relevant, as this is only called with service account credentials.

File: pybigquery/sqlalchemy_bigquery.py (hunk 1 above)
Comment (alonme) [reply pybigquery/sqlalchemy_bigquery.py]: Resolved

File: pybigquery/sqlalchemy_bigquery.py [hunk 3]
```diff
@@ -267,47 +268,61 @@ def __init__(
     def dbapi(cls):
         return dbapi
 
-    def create_connect_args(self, url):
-        location, dataset_id, arraysize, credentials_path, default_query_job_config = parse_url(url)
+    @staticmethod
+    def _add_default_dataset_to_job_config(job_config, project_id, dataset_id):
+        # If dataset_id is set, then we know the job_config isn't None
+        if dataset_id:
+            # If project_id is missing, use defualt project_id
```
Comment (tswast) [inline pybigquery/sqlalchemy_bigquery.py]:
```suggestion
            # If project_id is missing, use default project_id
```

**Comment by alonme** [comment]
@mxmzdlv Can you kindly merge and deploy a new version for this bug fix?

**Comment by mxmzdlv** [comment]
Thanks for fixing this!

**Pull request merged**